### ISSUE TITLE
Unspecified closure causes Why3 error

### ISSUE THREAD

**Issue by dewert99** [xldenis/creusot#525]
The following program:
```rust
use creusot_contracts::*;
use creusot_contracts::std::FnOnceSpec;

fn take_fn_once<A, B, F: FnOnce(A) -> B>(f: F) {}


pub fn test<T>() {
    take_fn_once(|s: T| {})
}
```
Gives the following why3 error:
```
File "target/debug/creusot_experiments-rlib.mlcfg", line 70, characters 13-20:
unbound function or predicate symbol 'resolve'
```
An easy workaround is to add requires true eg:
```rust
use creusot_contracts::*;
use creusot_contracts::std::FnOnceSpec;

fn take_fn_once<A, B, F: FnOnce(A) -> B>(f: F) {}


pub fn test<T>() {
    take_fn_once(#[requires(true)]|s: T| {})
}
```
which does work.

(This isn't high priority but I thought would it be good to make this visible)

**Comment by xldenis** [comment]
Argh, i thought I had fixed that recently...

**Comment by xldenis** [comment]
This was fixed by #516, I cannot replicate the issue any longer.
